Migrate to Manifest V3 and add cross-browser compatibility

scottmmcintyre opened the pull request:
Honestly I just threw this into Cursor and asked it to do it for me, seems to work alright when I load it as an unpacked extension in Chrome, I honestly didn't check it in Firefox but I'll see if I can

- Update manifest to Manifest V3
- Add browser-polyfill.js for cross-browser support
- Refactor newlove.user.js to use browser extension APIs
- Update copyright year
- Modernize JavaScript syntax
- Improve error handling and storage management

scottwiw committed:
Migrate to Manifest V3 and add cross-browser compatibility

- Update manifest to Manifest V3
- Add browser-polyfill.js for cross-browser support
- Refactor newlove.user.js to use browser extension APIs
- Update copyright year
- Modernize JavaScript syntax
- Improve error handling and storage management

scottwiw committed:
Prepare for Firefox and Chrome extension releases

- Update Makefile to build both Chrome and Firefox extension zips
- Include browser-polyfill.js in extension packages
- Bump script version to 2.0.0 for Firefox 109 compatibility
- Minor cleanup of browser-polyfill.js

iangreenleaf commented on newlove.user.js:
```diff
@@ -21,9 +21,10 @@
 // 1.2 - Erin [nichols] reworked this to work in Chrome!
 // 1.3 - Use native JSON for storage, fix problems with FF4
 // 1.4 - Updates for GreaseMonkey 3.0
+// 2.0 - Updates for Firefox 109
```
We probably shouldn't leave the AI in charge of this comment :smile:

iangreenleaf reviewed (commented):
Thanks! This is super helpful. My only concern would be testing it in Firefox to ensure it continues working there.  That's something I can probably do if you don't get to it, but it might take me a week or two. It looks like the AI might have retained compatibility with Greasemonkey, but it's hard to say for sure. I don't think the polyfill will get loaded in Greasemonkey, but it's probably also not needed.

It looks like it might now be possible to use it as a standalone Firefox extension, which could be a nice alternative to Greasemonkey. We'd need to test that too, though, before making it available.

scottmmcintyre commented:
I was able to test it as a standalone extension in Firefox Developer Edition, so that works at least

scottmmcintyre committed:
Update newlove.user.js

mcintyre1 committed:
Bunch of changes, moved old greasemoney context options to buttons, added ability to view old planlove

scottmmcintyre commented:
Added a bunch of new stuff, this one is a fairly significant change to the behavior/look so totally understand if you don't want to do all that. I realized I broke all of the context menu greasemonkey stuff but it was still in, so I re-added that functionality as buttons that appear up near the planlove checkbox, and also added a link to each author item to click and reveal hidden old plan love. I could have sworn the old extension let you do that but I didn't actually see that in the code? I guess I'm crazy. 

Here's a screenshot of the new stuff:

![image](https://github.com/user-attachments/assets/2373096a-74c3-4570-9ff7-2c3df1f7a264)
